polaris-core/src/main/java/org/apache/polaris/core/storage/StorageUtil.java

```diff
@@ -62,4 +68,50 @@ public class StorageUtil {
   public static @Nonnull String getBucket(URI uri) {
     return uri.getAuthority();
   }
+
+  /** Given a TableMetadata, extracts the locations where the table's [meta]data might be found. */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(TableMetadata tableMetadata) {
+    return getLocationsAllowedToBeAccessed(tableMetadata.location(), tableMetadata.properties());
+  }
+
+  /** Given a baseLocation and entity (table?) properties, extracts the relevant locations */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(
+      String baseLocation, Map<String, String> properties) {
+    Set<String> locations = new HashSet<>();
+    locations.add(baseLocation);
+    if (properties.containsKey(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_DATA_LOCATION_KEY)) {
+      locations.add(properties.get(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_DATA_LOCATION_KEY));
+    }
+    if (properties.containsKey(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_METADATA_LOCATION_KEY)) {
+      locations.add(
+          properties.get(IcebergTableLikeEntity.USER_SPECIFIED_WRITE_METADATA_LOCATION_KEY));
+    }
+    return removeRedundantLocations(locations);
+  }
+
+  /** Given a ViewMetadata, extracts the locations where the view's [meta]data might be found. */
+  public static @Nonnull Set<String> getLocationsAllowedToBeAccessed(ViewMetadata viewMetadata) {
+    return Set.of(viewMetadata.location());
+  }
+
+  /** Removes "redundant" locations, so {/a/b/, /a/b/c, /a/b/d} will be reduced to just {/a/b/} */
+  private static @Nonnull Set<String> removeRedundantLocations(Set<String> locationStrings) {
+    HashSet<String> result = new HashSet<>(locationStrings);
+    result.remove(null);
+
+    for (String potentialParent : locationStrings) {
+      for (String potentialChild : locationStrings) {
+        if (potentialParent != null
+            && potentialChild != null
```

Comment:
Shouldn't the elements never be `null`?

Reply:
Oh, just saw your other comment -- if we _remove_ null from the input list as you suggest, then yes this check can go away!